archive/archive_api/src/bag_request.py

```diff
@@ -0,0 +1,26 @@
+# -*- encoding: utf-8
+import daiquiri
+import json
+
+from werkzeug.exceptions import NotFound as NotFoundError
+
+logger = daiquiri.getLogger()
+
+
+def bag_request(dynamodb_resource, table_name, s3_client, bucket_name, id):
+    """
+    Fetch the contents of a Bag.
+    """
+
+    table = dynamodb_resource.Table(table_name)
+    item_response = table.get_item(Key={'id': id})
+
+    try:
+        response = s3_client.get_object(
+            Bucket=bucket_name,
+            Key=item_response['Item']['s3key']
+        )
+    except KeyError:
+        raise NotFoundError(f'No bag found for id={id!r}')
```

Comment:
Minor: We shouldn't be raising a Werkzeug error here, because it's very HTTP specific – it's better to raise a regular Python exception, and catch/re-raise it in the context of an HTTP Flask app.

Reply:
@alexwlchan Copied from existing report_ingest_status.py code

Reply:
Yes, I think we should keep the http errors in the api layer.  I may have written it originally in the `report_ingest_status.py` code feel free to change it.

Reply:
I'll change the other files in a separate commit. Thanks.